packages/fast-components-react-base/src/button/button.tsx

```diff
@@ -37,19 +45,52 @@ class Button extends Foundation<IButtonHandledProps & IManagedClasses<IButtonCla
      * Generates class names
      */
     protected generateClassNames(): string {
-        return super.generateClassNames(get(this.props, "managedClasses.button"));
+        let classLocation: string;
+
+        if (this.props.primary) {
+            classLocation = "managedClasses.button_primary";
+        } else if (this.props.outline) {
+            classLocation = "managedClasses.button_outline";
+        } else if (this.props.lightweight) {
+            classLocation = "managedClasses.button_lightweight";
+        } else if (this.props.justified) {
+            classLocation = "managedClasses.button_justified";
+        } else {
+            classLocation = "managedClasses.button";
+        }
+
+        return super.generateClassNames(get(this.props, classLocation));
     }
 
     /**
-     * Creates tags for rendering based on href
+     * Stores HTML tag for use in render
      */
-    private generateHTMLTag(): string {
-        switch (this.props.tag) {
-            case ButtonHTMLTags.a:
-                return "a";
-            case ButtonHTMLTags.button:
-            default:
-                return "button";
+    private renderDisabled(): any {
+        if (this.props.disabled) {
+            if (this.tag === ButtonHTMLTags.a) {
+                return {"aria-disabled": true};
+            } else {
+                return {"disabled": true};
+            }
```

Comment:
this could be `return this.tag === ButtonHTMLTags.a ? {"aria-disabled": true} : {"disabled": true};`